compiler/src/iree/compiler/Codegen/Utils/GPUUtils.cpp

```diff
@@ -522,28 +522,31 @@ Value getCombiningIdentityValue(Location loc, OpBuilder &builder,
 }
 
 /// Return a matching GPU reduction operations.
-static std::optional<gpu::AllReduceOperation>
+static gpu::AllReduceOperation
 combiningKindToAllReduce(vector::CombiningKind kind) {
-  using gpu::AllReduceOperation;
-  using vector::CombiningKind;
-
-  switch (kind) {
-  case CombiningKind::ADD:
-    return AllReduceOperation::ADD;
-  case CombiningKind::AND:
-    return AllReduceOperation::AND;
-  case CombiningKind::MUL:
-    return AllReduceOperation::MUL;
-  case CombiningKind::OR:
-    return AllReduceOperation::OR;
-  case CombiningKind::XOR:
-    return AllReduceOperation::XOR;
-  // Currently, the min/max reductions are not well-defined in the gpu dialect.
-  // See https://github.com/llvm/llvm-project/issues/72354.
-  default:
-    break;
-  }
-  return std::nullopt;
+  switch (unsigned(kind)) {
```

Comment:
Why do we need unsigned cast?

Reply:
I think this was originally because I was getting some weird compiler warning, but removing the casts doesn't stop it building, so I'll do that.